### DISCUSSION TITLE
Contributing to svelte-kit, running svelte-kit from source

### DISCUSSION THREAD

**Discussion by joshamaju** [sveltejs/kit#9581] [Q&A]
I cloned the svelte-kit repository and I can't for the life of me figure out where it all begins. I've looked through the package.json for each package and can't find a build script or a main entry file. The kit package has svelte-kit.js, but that just points to the cli.js file.

I guessed the build was run through some action on GitHub, checked that, but it's just a reference to changeset publish; which is another blackhole of its own.

I also tried to scrutinise the vite config, but came up empty. Tried running the test, they're too much and don't really point to any where concrete.

I need help, can someone help me figure this out.

**Comment by vicentematus** [comment]
Hey, the place to start normally in a lot of open source of repositories is to check on the root directory the `CONTRIBUTING.MD`. Take a look for [Svelte Kit here.](https://github.com/sveltejs/kit/blob/6b6f4565c9f1486cb43e921113f145e001078090/CONTRIBUTING.md)

A few questions: did you install pnpm? did you installed the packages? 

> I guessed the build was run through some action on GitHub, checked that, but it's just a reference to changeset publish; which is another blackhole of its own.

Whenever an adapter changes, you need to publish a change set like it says on the CONTRIBUTING.MD file:
```
Generating changelogs
For changes to be reflected in package changelogs, run pnpm changeset and follow the prompts.
``` 


And all the CI actions of precommit hooks as you call it are referenced here:

```
To use the git hooks in the repo, which will save you from waiting for CI to tell you that you forgot to lint, run this:

git config core.hookspath .githooks
``` 
Which it will run all the CI commands before commiting on the official repository. Like the linter, the formatter, and the test.

And if you want to set up the kit.svelte.dev and the docs file you go to the directory and run: 

```
cd sites/kit.svelte.dev/ 
pnpm run build
``` 

After that you are set to go.

There some few things that one need to figure it out like the kit.svelte.dev doc thing, but we can always improve the documentation.

**Reply by joshamaju** [reply]
Thanks for your response. I did go through the contributing markdown file. I do use `pnpm` and installed the packages. I checked the changeset reference and their docs before positing my question and everything I found on that was on publishing. But I'm not looking to publishing, at least not yet.

What I want to achieve is to be able to make changes local and test those changes locally. I figured the tests would be one place to start. But there are too many tests, even switching to each package folder still runs about 638 tests. Which makes it difficult to really see what I'm doing.

I can't find one single build command script, I figured it has something to do with the vite setup. But also didn't find anything on that.

The sandbox example mentioned in the doc didn't work either, i got

```
  ERROR
    Invalid command: dev

  Run `$ svelte-kit --help` for more info.
```

After examining the codebase, I understand that the `kit` package is where most of the work happens. But how to get anything started is a mystery at this point.

**Reply by vicentematus** [reply]
Are you refering to how to start the project? like npm run dev? if  thats the case Svelte Kit is a mono repo which contain different projects. 

To start svelte kit standalone you need to be on the root folder of the repository and run `pnpm run start`. For more info or reference you can check the `package.json` (of the root folder) and see the scripts. I hope that's the issue.
